src/mscorlib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -21,46 +21,30 @@ internal static partial class AsyncMethodBuilder
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.stateMachine);
             }
 
-            // enregistrer variables with 0 post-fix so they can be used in registers without EH forcing them to stack
-            // Capture references to Thread Contexts
-            Thread currentThread0 = Thread.CurrentThread;
-            Thread currentThread = currentThread0;
-            ExecutionContext previousExecutionCtx0 = currentThread0.ExecutionContext;
+            Thread currentThread = Thread.CurrentThread;
```

Comment:
There is no Thread.CurentThread in CoreRT CoreLib today - the whole thing is structured quite differently. I think this file should be removed from the shared CoreLib partition. There is not much to share for now.

Reply:
How does `AsyncMethodBuilderCore.Start<TStateMachine>(ref TStateMachine stateMachine)` work in corert as it uses `Thread.CurrentThread`?

https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs#L724-L733

Reply:
That file has `using Thread = Internal.Runtime.Augments.RuntimeThread;` at the top.

Reply:
You maybe do the same here.